HTWDD/Components/Grades/Models/GradeDataSource.swift

```diff
@@ -79,9 +96,12 @@ class GradeDataSource: CollectionViewDataSource {
                 })
             .disposed(by: self.disposeBag)
     }
-
-    func information(`for` section: Int) -> GradeService.Information {
-        return self.semesters[section]
+ 
+    func information(`for` section: Int) -> Either<GradeService.Information, String> {
+        guard section > 0 else {
+            return .right("Gesamtdurchschnitt")
```

Comment:
Use localised String instead 👻
Although I think the overall grade average cell doesn't need a header at all.